db/migrate/20190220191436_create_shortened_urls_table.rb

```diff
@@ -0,0 +1,34 @@
+class CreateShortenedUrlsTable < ActiveRecord::Migration[4.2]
+  def change
+    create_table :shortened_urls do |t|
+      # we can link this to a user for interesting things
+      t.integer :owner_id
+      t.string :owner_type, limit: 20
+
+      # the real url that we will redirect to
+      t.text :url, null: false, length: 2083
+
+      # the unique key
+      t.string :unique_key, limit: 10, null: false
+
+      # a category to help categorize shortened urls
+      t.string :category
+
+      # how many times the link has been clicked
+      t.integer :use_count, null: false, default: 0
+
+      # valid until date for expirable urls
+      t.datetime :expires_at
```

Comment:
Cool! How are expired urls cleaned? Is there a rake task or something for that?

Reply:
Expiry is optional on creation of a short url. Do you think we should add it? I didn't see any reason why.

Reply:
Just clean-up... I guess we can do it, if it becomes very popular.